e2etest/layer2_test.go

```diff
@@ -257,21 +259,31 @@ var _ = ginkgo.Describe("L2", func() {
 			})
 		framework.ExpectNoError(err)
 
-		events, err := cs.CoreV1().Events(namespace).List(context.Background(), metav1.ListOptions{FieldSelector: "reason=nodeAssigned"})
-		framework.ExpectNoError(err)
+		Eventually(func() error {
+			events, err := cs.CoreV1().Events(namespace).List(context.Background(), metav1.ListOptions{FieldSelector: "reason=nodeAssigned"})
+			framework.ExpectNoError(err)
```

Comment:
I think you should return the err instead

Reply:
Being able to list nodes should not depend on the timing, that's why I left it that way.

Reply:
IIUC there could be a scenario where listing the events fails due to some flake/connection error but in the next iteration of the Eventually it would have worked - if so do we want to fail the test because of that?

Reply:
My idea was that it was less likely to happen (compared to the real reason why we have flakes), but changing it won't hurt.

Reply:
Done